index.js

```diff
@@ -17,12 +17,25 @@ function shouldComponentUpdate(nextProps, nextState) {
   return !shallowEqual(this.props, nextProps) || !shallowEqual(this.state, nextState);
 }
 
+function getFunctionName(fun) {
+  var ret = fun.toString();
+  ret = ret.substr('function '.length);
+  ret = ret.substr(0, ret.indexOf('('));
+  return ret;
+}
+
 /**
  * Makes the given component "pure".
  *
  * @param object component Component.
  */
 function pureRenderDecorator(component) {
+  if (component.prototype.shouldComponentUpdate !== undefined) {
+    var componentName = component.prototype.name || component.name || getFunctionName(component);
+    throw new Error("Cannot add a pure render decorator on a component that already"
```

Comment:
The error message seems repetitive, I would change it to something like 
```js
"Cannot add a pure render decorator to " + componentName + ", because it already implements `shouldComponentUpdate()`."
```